File: Lecture 4: Solving Pattern Questions (Part-2)/all_patterns.cpp
#include <iostream>

using namespace std;

void pattern1();
void pattern2();
void pattern3();
void pattern4();
void pattern5();
void pattern6();
void pattern7();
void pattern8();
void pattern9();
void pattern10();
void pattern11();
void pattern12();
void pattern13();
void pattern14();
void pattern15();

int main()
{
    int no_of_pattern;
    cout << endl;
    cout << "Select which pattern you want to print from below list:\n";
    cout << endl;
    cout << "1. Square Pattern." << endl;
    cout << endl;
    cout << "2. Number Square Pattern." << endl;
    cout << endl;
    cout << "3. Another Number Square Pattern." << endl;
    cout << endl;
    cout << "4. Inverted Number Square Pattern." << endl;
    cout << endl;
    cout << "5. Counting Square Pattern." << endl;
    cout << endl;
    cout << "6. 90 Degree Triangle Pattern." << endl;
    cout << endl;
    cout << "7. 90 Degree Counting Triangle Pattern." << endl;
    cout << endl;
    cout << "8. 90 Degree Inverted Counting Triangle Pattern." << endl;
    cout << endl;
    cout << "9. Alphabetic Square Pattern." << endl;
    cout << endl;
    cout << "10. Alphabetic Triangle Pattern." << endl;
    cout << endl;
    cout << "11. Inverted Alphabetic Triangle Pattern." << endl;
    cout << endl;
    cout << "12. Right Angle Inverted Character Triangle Pattern." << endl;
    cout << endl;
    cout << "13. Right Angle Reverse Character Triangle Pattern." << endl;
    cout << endl;
    cout << "14. Number Pyramid Pattern." << endl;
    cout << endl;
    cout << "15. Dabangg Pattern." << endl;
    cout << endl;
    cin >> no_of_pattern;
    cout << endl;

    switch (no_of_pattern)
    {
    case 1:
        pattern1();
        break;

    case 2:
        pattern2();
        break;

    case 3:
        pattern3();
        break;

    case 4:
        pattern4();
        break;

    case 5:
        pattern5();
        break;

    case 6:
        pattern6();
        break;

    case 7:
        pattern7();
        break;

    case 8:
        pattern8();
        break;

    case 9:
        pattern9();
        break;

    case 10:
        pattern10();
        break;

    case 11:
        pattern11();
        break;

    case 12:
        pattern12();
        break;

    case 13:
        pattern13();
        break;

    case 14:
        pattern14();
        break;

    case 15:
        pattern15();
        break;

    default:
        cout << "Invalid Number, please select a number from above given list :)" << endl;
        cout << endl;
    }

    return 0;
}

void pattern1()
{
    // Square Pattern
    cout << endl;
    cout << "You are going to see a square pattern:- " << endl;

    int side, i, j;
    cout << endl;
    cout << "Enter side of the square:";
    cout << endl;
    cin >> side;
    cout << endl;

    for (i = 1; i <= side; i++)
    {
        for (j = 1; j <= side; j++)
        {
            cout << "* ";
        }
        cout << endl;
    }
    cout << endl;
}

void pattern2()
{
    // Number Square Pattern
    cout << endl;
    cout << "You are going to see a number square pattern:- " << endl;

    int side, i, j;
    cout << endl;
    cout << "Enter side of the square:";
    cout << endl;
    cin >> side;
    cout << endl;

    for (i = 1; i <= side; i++)
    {
        for (j = 1; j <= side; j++)
        {
            cout << i << "  ";
        }
        cout << endl;
    }
    cout << endl;
}

void pattern3()
{
    // Number Square Pattern - Part 2
    cout << endl;
    cout << "You are going to see another version of number square pattern:- " << endl;

    int side, i, j;
    cout << endl;
    cout << "Enter side of the square:";
    cout << endl;
    cin >> side;
    cout << endl;

    for (i = 1; i <= side; i++)
    {
        for (j = 1; j <= side; j++)
        {
            cout << j << "  ";
        }
        cout << endl;
    }
    cout << endl;
}

void pattern4()
{
    // Inverted Number Square Pattern
    cout << endl;
    cout << "You are going to see inverted number square pattern:- " << endl;

    int side, i, j;
    cout << endl;
    cout << "Enter side of the square:";
    cout << endl;
    cin >> side;
    cout << endl;

    for (i = 1; i <= side; i++)
    {
        for (j = 1; j <= side; j++)
        {
            cout << (side - j + 1) << "  ";
        }
        cout << endl;
    }
    cout << endl;
}

void pattern5()
{
    // Counting Square Pattern
    cout << endl;
    cout << "You are going to see counting square pattern:- " << endl;

    int side, i, j, counter = 1;
    cout << endl;
    cout << "Enter side of the square:";
    cout << endl;
    cin >> side;
    cout << endl;

    for (i = 1; i <= side; i++)
    {
        for (j = 1; j <= side; j++)
        {
            cout << counter << "  ";
            counter++;
        }
        cout << endl;
    }
    cout << endl;
}

void pattern6()
{
    // Right Angled Triangle Pattern
    cout << endl;
    cout << "You are going to see right angled triangle pattern:- " << endl;

    int height, i, j;
    char ch;
    cout << endl;
    cout << "Enter height of the triangle:";
    cout << endl;
    cin >> height;
    cout << endl;
    cout << "Enter the symbol of which you want to create the triangle also:- ";
    cout << endl;
    cin >> ch;
    cout << endl;

    for (i = 1; i <= height; i++)
    {
        for (j = 1; j <= i; j++)
        {
            cout << ch << "  ";
        }
        cout << endl;
    }
    cout << endl;
}

void pattern7()
{
    // Right Angled Counting Triangle Pattern
    cout << endl;
    cout << "You are going to see right angled triangle pattern:- " << endl;

    int height, i, j, k, counting = 1;
    char ch;
    cout << endl;
    cout << "Enter height of the triangle:";
    cout << endl;
    cin >> height;
    cout << endl;
    cout << "Enter the symbol of which you want to create the triangle:- ";
    cout << endl;
    cin >> ch;
    cout << endl;

    for (i = 1; i <= height; i++)
    {
        int value = i;
        for (j = 1; j <= i; j++)
        {
            cout << value << " ";
            value++;
        }
        cout << endl;
    }
    cout << endl;
}

void pattern8()
{
    // Right Angled Inverted Counting Triangle Pattern
    cout << endl;
    cout << "You are going to see right angled inverted triangle pattern:- " << endl;

    int height, i, j, val, k;
    char ch;
    cout << endl;
    cout << "Enter height of the triangle:";
    cout << endl;
    cin >> height;
    cout << endl;

    for (i = 1; i <= height; i++)
    {
        for (j = 1; j <= i; j++)
        {
            cout << i - j + 1 << " ";
        }
        cout << endl;
    }
    cout << endl;
}

void pattern9()
{
    // Alphabetic Square Pattern
    cout << endl;
    cout << "You are going to see alphabetic square pattern:- " << endl;

    int side, i, j;
    char ch = 'A'; //* (For printing same alphabets in the rows)
    cout << endl;
    cout << "Enter side of the square:-";
    cout << endl;
    cin >> side;
    cout << endl;

    for (i = 1; i <= side; i++)
    {
        //! char ch = 'A';   (For Printing continuous aplhabetic order)
        for (j = 1; j <= side; j++)
        {
            cout << ch << " ";
            //! ch++;
        }
        cout << endl;
        ch++;
    }
    cout << endl;
}

void pattern10()
{
    // Alphabetic Triangle Pattern
    cout << endl;
    cout << "You are going to see alphabetic triangle pattern:- " << endl;

    int height, i, j;
    char ch = 'A'; //* (For printing same alphabets in the rows)
    cout << endl;
    cout << "Enter height of the triangle:-";
    cout << endl;
    cin >> height;
    cout << endl;

    for (i = 1; i <= height; i++)
    {
        //! char ch = 'A'; // (For Printing continuous aplhabetic order)
        for (j = 1; j <= i; j++)
        {
            cout << ch << " ";
            //! ch++; // (For Printing continuous aplhabetic order)
        }
        cout << endl;
        ch++; //* (For printing same alphabets in the rows)
    }
    cout << endl;
}

void pattern11()
{
    // Inverted Alphabetic Triangle Pattern
    cout << endl;
    cout << "You are going to see inverted alphabetic triangle pattern:- " << endl;

    int height, i, j;
    char value, ch = 'A';
    cout << endl;
    cout << "Enter height of the triangle:-";
    cout << endl;
    cin >> height;
    cout << endl;

    for (i = 1; i <= height; i++)
    {
        value = ch + height - i;
        for (j = 1; j <= i; j++)
        {
            cout << value << "  ";
            value++;
        }
        cout << endl;
    }
    cout << endl;
}

void pattern12()
{
    // Inverted Right Angle Character Triangle Pattern
    cout << endl;
    cout << "You are going to see inverted right angle triangle character pattern:- " << endl;

    int height, i, j, k;
    char ch;
    cout << endl;
    cout << "Enter height of the triangle:";
    cout << endl;
    cin >> height;
    cout << endl;
    cout << "Enter the symbol of which you want to create the triangle:- ";
    cout << endl;
    cin >> ch;
    cout << endl;

    for (i = 1; i <= height; i++)
    {
        for (k = 1; k <= (height - i); k++)
        {
            cout << "  ";
        }
        for (j = 1; j <= i; j++)
        {
            cout << ch << " ";
        }
        cout << endl;
    }
}

void pattern13()
{
    // Reverse Right Angle Character Triangle Pattern
    cout << endl;
    cout << "You are going to see reverse right angle triangle character pattern:- " << endl;

    int height, i, j;
    char ch;
    cout << endl;
    cout << "Enter height of the triangle:- ";
    cout << endl;
    cin >> height;
    cout << endl;
    cout << "Enter the symbol of which you want to create the triangle:- ";
    cout << endl;
    cin >> ch;
    cout << endl;

    for (i = 1; i <= height; i++)
    {
        /* for (j = 1; j <= (height - i + 1); j++)
        {
            cout << ch << " ";
        } */
        for (j = height; j >= i; j--)
        {
            cout << ch << " ";
        }
        cout << endl;
    }
}

void pattern14()
{
    // Number Pyramid Pattern
    cout << endl;
    cout << "You are going to see a number pyramid pattern:- " << endl;

    int height, i, j, k, l;
    cout << endl;
    cout << "Enter height of the pyramid:- ";
    cout << endl;
    cin >> height;
    cout << endl;

    for (i = 1; i <= height; i++)
    {
        for (k = 1; k <= (height - i); k++)
        {
            cout << "  ";
        }
        for (j = 1; j <= i; j++)
        {
            cout << j << " ";
        }
        for (l = i - 1; l >= 1; l--)
        {
            cout << l << " ";
        }
        cout << endl;
    }
}

void pattern15()
{
    // Dabangg Pattern
    cout << endl;
    cout << "You are going to see dabangg pattern:- " << endl;

    int height, i, j, k, l;
    cout << endl;
    cout << "Enter height of the pattern:- ";
    cout << endl;
    cin >> height;
    cout << endl;

    for (i = 1; i <= height; i++)
    {
        for (j = 1; j <= (height - i + 1); j++)
        {
            cout << j << " ";
        }
        for (k = 1; k <= (i - 1); k++)
        {
            cout << "* ";
        }
        for (k = (i - 1); k >= 1; k--)
        {
            cout << "* ";
        }
        for (l = (height - i + 1); l >= 1; l--)
        {
            cout << l << " ";
        }
        cout << endl;
    }
}

File: Lecture004 Solving Pattern Questions (Part-2)/all_patterns.java
import java.util.Scanner;

class AllPatterns {
    public static void main(String[] args) {

        int no_of_pattern;
        System.out.println("Select which pattern you want to print from below list:");
        System.out.println("1. Square Pattern.");
        System.out.println("2. Number Square Pattern.");
        System.out.println("3. Another Number Square Pattern.");
        System.out.println("4. Inverted Number Square Pattern.");
        System.out.println("5. Counting Square Pattern.");
        System.out.println("6. 90 Degree Triangle Pattern.");
        System.out.println("7. 90 Degree Counting Triangle Pattern.");
        System.out.println("8. 90 Degree Inverted Counting Triangle Pattern.");
        System.out.println("9. Alphabetic Square Pattern.");
        System.out.println("10. Alphabetic Triangle Pattern.");
        System.out.println("11. Inverted Alphabetic Triangle Pattern.");
        System.out.println("12. Right Angle Inverted Character Triangle Pattern.");
        System.out.println("13. Right Angle Reverse Character Triangle Pattern.");
        System.out.println("14. Number Pyramid Pattern.");
        System.out.println("15. Dabangg Pattern.");
        System.out.print("Enter your choice: ");

        Scanner sc = new Scanner(System.in);
        no_of_pattern = sc.nextInt();

        switch (no_of_pattern) {
            case 1:
                pattern1(sc);
                break;

            case 2:
                pattern2(sc);
                break;

            case 3:
                pattern3(sc);
                break;

            case 4:
                pattern4(sc);
                break;

            case 5:
                pattern5(sc);
                break;

            case 6:
                pattern6(sc);
                break;

            case 7:
                pattern7(sc);
                break;

            case 8:
                pattern8(sc);
                break;

            case 9:
                pattern9(sc);
                break;

            case 10:
                pattern10(sc);
                break;

            case 11:
                pattern11(sc);
                break;

            case 12:
                pattern12(sc);
                break;

            case 13:
                pattern13(sc);
                break;

            case 14:
                pattern14(sc);
                break;

            case 15:
                pattern15(sc);
                break;

            default:
                System.out.println("Invalid Number, please select a number from above given list :)");
                System.out.println();
        }

        sc.close();
    }

    static void pattern1(Scanner sc) {
        // Square Pattern
        System.out.println();
        System.out.println("You are going to see a square pattern:- ");
        System.out.print("Enter side of the square: ");

        int side, i, j;
        side = sc.nextInt();
        System.out.println();
        for (i = 1; i <= side; i++) {
            for (j = 1; j <= side; j++) {
                System.out.print("* ");
            }
            System.out.println();
        }
        System.out.println();
    }

    static void pattern2(Scanner sc) {
        // Number Square Pattern
        System.out.println();
        System.out.println("You are going to see a square pattern:- ");
        System.out.print("Enter side of the square: ");

        int side, i, j;
        side = sc.nextInt();
        System.out.println();
        for (i = 1; i <= side; i++) {
            for (j = 1; j <= side; j++) {
                System.out.print(i + " ");
            }
            System.out.println();
        }
        System.out.println();
    }

    static void pattern3(Scanner sc) {
        // Number Square Pattern - Part 2
        System.out.println();
        System.out.println("You are going to see another version of number square pattern:- ");

        int side, i, j;
        System.out.print("Enter side of the square: ");
        side = sc.nextInt();

        for (i = 1; i <= side; i++) {
            for (j = 1; j <= side; j++) {
                System.out.print(j + " ");
            }
            System.out.println();
        }
        System.out.println();
    }

    static void pattern4(Scanner sc) {
        // Inverted Number Square Pattern
        System.out.println();
        System.out.println("You are going to see inverted number square pattern:- ");

        int side, i, j;
        System.out.print("Enter side of the square: ");
        side = sc.nextInt();
        System.out.println();

        for (i = 1; i <= side; i++) {
            for (j = 1; j <= side; j++) {
                System.out.print((side - j + 1) + " ");
            }
            System.out.println();
        }
        System.out.println();
    }

    static void pattern5(Scanner sc) {
        // Counting Square Pattern
        System.out.println();
        System.out.println("You are going to see counting square pattern:- ");

        int side, i, j, counter = 1;
        System.out.println();
        System.out.print("Enter side of the square: ");
        side = sc.nextInt();
        System.out.println();

        for (i = 1; i <= side; i++) {
            for (j = 1; j <= side; j++) {
                System.out.print(counter + " ");
                counter++;
            }
            System.out.println();
        }
        System.out.println();
    }

    static void pattern6(Scanner sc) {
        // Right Angled Triangle Pattern
        System.out.println();
        System.out.println("You are going to see right angled triangle pattern:- ");

        int height, i, j;
        char ch;
        System.out.println();
        System.out.print("Enter height of the triangle: ");
        height = sc.nextInt();

        System.out.print("Enter the symbol of which you want to create the triangle also: ");
        ch = sc.next().charAt(0);
        System.out.println();

        for (i = 1; i <= height; i++) {
            for (j = 1; j <= i; j++) {
                System.out.print(ch + " ");
            }
            System.out.println();
        }
        System.out.println();
    }

    static void pattern7(Scanner sc) {
        // Right Angled Counting Triangle Pattern
        System.out.println();
        System.out.println("You are going to see right angled triangle pattern:- ");

        int height;
        char ch;

        System.out.print("Enter height of the triangle: ");
        height = sc.nextInt();

        System.out.print("Enter the symbol of which you want to create the triangle: ");
        ch = sc.next().charAt(0);
        System.out.println();

        for (int i = 1; i <= height; i++) {
            for (int j = 1; j <= i; j++) {
                System.out.print(ch + " ");
            }
            System.out.println();
        }
        System.out.println();
    }

    static void pattern8(Scanner sc) {
        // Right Angled Inverted Counting Triangle Pattern
        System.out.println();
        System.out.println("You are going to see right angled inverted triangle pattern:-");

        int height, i, j;
        System.out.println();
        System.out.print("Enter height of the triangle: ");
        height = sc.nextInt();
        System.out.println();

        for (i = 1; i <= height; i++) {
            for (j = 1; j <= i; j++) {
                System.out.print((i - j + 1) + " ");
            }
            System.out.println();
        }
        System.out.println();
    }

    static void pattern9(Scanner sc) {
        // Alphabetic Square Pattern
        System.out.println();
        System.out.println("You are going to see alphabetic square pattern:- ");

        int side, i, j;
        char ch = 'A'; // * (For printing same alphabets in the rows)
        System.out.println();
        System.out.print("Enter side of the square: ");
        side = sc.nextInt();
        System.out.println();

        for (i = 1; i <= side; i++) {
            // ! char ch = 'A'; (For Printing continuous aplhabetic order)
            for (j = 1; j <= side; j++) {
                System.out.print(ch + " ");
                // ! ch++;
            }
            System.out.println();
            ch++;
        }
        System.out.println();
    }

    static void pattern10(Scanner sc) {
        // Alphabetic Triangle Pattern
        System.out.println();
        System.out.println("You are going to see alphabetic triangle pattern:- ");

        int height, i, j;
        char ch = 'A'; // * (For printing same alphabets in the rows)
        System.out.println();
        System.out.print("Enter height of the triangle: ");
        height = sc.nextInt();
        System.out.println();

        for (i = 1; i <= height; i++) {
            // ! char ch = 'A'; // (For Printing continuous aplhabetic order)
            for (j = 1; j <= i; j++) {
                System.out.print(ch + " ");
                // ! ch++; // (For Printing continuous aplhabetic order)
            }
            System.out.println();
            ch++; // * (For printing same alphabets in the rows)
        }
        System.out.println();
    }

    static void pattern11(Scanner sc) {
        // Inverted Alphabetic Triangle Pattern
        System.out.println();
        System.out.println("You are going to see inverted alphabetic triangle pattern:- ");

        int height, i, j;
        char value, ch = 'A';
        System.out.println();
        System.out.print("Enter height of the triangle: ");
        height = sc.nextInt();
        System.out.println();

        for (i = 1; i <= height; i++) {
            value = (char) (ch + height - i);
            for (j = 1; j <= i; j++) {
                System.out.print(value + " ");
                value++;
            }
            System.out.println();
        }
        System.out.println();
    }

    static void pattern12(Scanner sc) {
        // Inverted Right Angle Character Triangle Pattern
        System.out.println();
        System.out.println("You are going to see inverted right angle triangle character pattern:- ");

        int height, i, j, k;
        char ch;
        System.out.print("Enter height of the triangle: ");
        height = sc.nextInt();

        System.out.print("Enter the symbol of which you want to create the triangle: ");
        ch = sc.next().charAt(0);
        System.out.println();

        for (i = 1; i <= height; i++) {
            for (k = 1; k <= (height - i); k++) {
                System.out.print("  ");
            }
            for (j = 1; j <= i; j++) {
                System.out.print(ch + " ");
            }
            System.out.println();
        }
    }

    static void pattern13(Scanner sc) {
        // Reverse Right Angle Character Triangle Pattern
        System.out.println();
        System.out.println("You are going to see reverse right angle triangle character pattern:- ");

        int height, i, j;
        char ch;
        System.out.println();
        System.out.print("Enter height of the triangle: ");
        height = sc.nextInt();

        System.out.print("Enter the symbol of which you want to create the triangle: ");
        ch = sc.next().charAt(0);
        System.out.println();

        for (i = 1; i <= height; i++) {
            /*
             * for (j = 1; j <= (height - i + 1); j++)
             * {
             * cout << ch << " ";
             * }
             */
            for (j = height; j >= i; j--) {
                System.out.print(ch + " ");
            }
            System.out.println();
        }
    }

    static void pattern14(Scanner sc) {
        // Number Pyramid Pattern
        System.out.println();
        System.out.println("You are going to see a number pyramid pattern:- ");

        int height, i, j, k, l;
        System.out.println();
        System.out.print("Enter height of the pyramid: ");
        height = sc.nextInt();
        System.out.println();

        for (i = 1; i <= height; i++) {
            for (k = 1; k <= (height - i); k++) {
                System.out.print("  ");
            }

            for (j = 1; j <= i; j++) {
                System.out.print(j + " ");
            }

            for (l = i - 1; l >= 1; l--) {
                System.out.print(l + " ");
            }

            System.out.println();
        }
    }

    static void pattern15(Scanner sc) {
        // Dabangg Pattern
        System.out.println();
        System.out.println("You are going to see dabangg pattern:- ");

        int height, i, j, k, l;
        System.out.println();
        System.out.print("Enter height of the pattern: ");
        height = sc.nextInt();
        System.out.println();

        for (i = 1; i <= height; i++) {
            for (j = 1; j <= (height - i + 1); j++) {
                System.out.print(j + " ");
            }

            for (k = 1; k <= (i - 1); k++) {
                System.out.print("* ");
            }

            for (k = (i - 1); k >= 1; k--) {
                System.out.print("* ");
            }

            for (l = (height - i + 1); l >= 1; l--) {
                System.out.print(l + " ");
            }

            System.out.println();
        }
    }

}